### PR TITLE
Metro mode selector - more a11y improvements

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
Splits each mode button into a checkbox (with label) + a dropdown button that triggers the popup.
As a result, removes the `disableHover` and related features, and cleans up CSS a good bit.

The following functionality still applies:
- Hovering over the label for a checked mode still triggers the popup.
- Checking a mode for the first time triggers the popup.
- Mobile: first click to check a mode triggers the popup. Also, a long-press on a checked mode label still triggers the popup (at least on Chrome on Android).
- Keyboard: activating the label only checks or unchecks the mode. To trigger the popup, press TAB to reveal the dropdown button, and press that to trigger the popup. To dismiss the popup, press ESC (a floating-ui/react feature), or Shift+TAB and press the dropdown button again.

### CODE REVIEW

File: packages/trip-form/src/MetroModeSelector/index.tsx [hunk 1]
```diff
@@ -139,76 +194,89 @@ function ModeButton({
...
+          <FormattedMessage
+            defaultMessage={defaultMessages["otpUi.ModeSelector.settingsLabel"]}
+            description="Label for the button to open settings for a travel mode."
+            id="otpUi.ModeSelector.settingsLabel"
+            values={{ mode: modeButton.label }}
+          />
+        </InvisibleA11yLabel>
+      </button>
       {renderDropdown && (
         <FloatingFocusManager context={context} modal={false}>
```
Comment (daniel-heppner-ibigroup) [inline packages/trip-form/src/MetroModeSelector/index.tsx:268]: Adjusting `FloatingFocusManager` like this fixes the issue of the buttons gaining focus when the popup closes, and it also traps the tab focus inside the popup until `esc` is pressed. We could also add a dismiss button that is hidden visually but can be tabbed to, if `esc` is not good enough.

```suggestion
        <FloatingFocusManager context={context} returnFocus={false}>
```

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Thank you so much for this! It's so slick. I made a suggestion which I think fixes the focus issue, let me know what you think about that. 

It works great on mobile and touch interfaces too, I love how smooth it is. Thank you!!

File: packages/trip-form/src/MetroModeSelector/index.tsx (hunk 1 above)
Comment (binh-dam-ibigroup) [reply packages/trip-form/src/MetroModeSelector/index.tsx:268]: Good point on keeping the focus within the popup. Maybe it should just be `<FloatingFocusManager context={context}>`, as we want to return the focus to the parent button when dismissing the popup.

Comment (binh-dam-ibigroup) [reply packages/trip-form/src/MetroModeSelector/index.tsx]: See https://github.com/ibi-group/otp-ui/pull/13/commits/c80bbdd2cca1af487502964e9f2442169c753ebf for updates. Feel free to merge into your PR branch when you are ready and have no other comments.

Comment (daniel-heppner-ibigroup) [reply packages/trip-form/src/MetroModeSelector/index.tsx]: Do we need it to return the focus? The focus ring appearing when you leave the hover element could be confusing for folks. I tested it with keyboard with `returnFocus=false` and it seemed like focus still went back to where it should, but maybe I missed a use case?

Comment (daniel-heppner-ibigroup) [reply packages/trip-form/src/MetroModeSelector/index.tsx]: Slight difference: it doesn't refocus the mode selector upon hitting `esc`, but when you press tab again it starts from the correct spot.

Comment (daniel-heppner-ibigroup) [reply packages/trip-form/src/MetroModeSelector/index.tsx]: If after reading this you still believe we should have returnFocus as true, please merge this! No need for more discussion.

Comment (binh-dam-ibigroup) [reply packages/trip-form/src/MetroModeSelector/index.tsx]: I think we should **show**, visually, a focused element when dismissing the popup using ESC. That way keyboard users know where they are. It looks like with `returnFocus={false}`, when dismissing the popup, the focus might remain within the `<FloatingFocusManager>`, which contents would be invisible.

**Pull request merged**